Add backoff config to loki source file (#4927)
grafana/agent: docs/sources/flow/reference/components/loki.source.file.md

The component can only support one compression format at a time, in order to
handle multiple formats, you will need to create multiple components.

> ### backoff block

Reviewer: Why call it backoff?

Is there a reason to deviate from promtails naming?

![image](https://github.com/grafana/agent/assets/17101802/03a3295a-a35d-411c-b692-a7388d521adb)

Author: I just followed the naming suggested by @rfratto in the issue. What was the initial reason to change the naming Robert?

Reviewer 2: Is `backoff` a predictable or understandable block name in context? As an outsider to the nuances of why the names are chosen, `file_watch` is more relatable to the function than `backoff`.

Reviewer 3: Good question. I'm so used to seeing "backoff settings" w/r/t scraping and sending data that I don't know if I have a bias here. 

Either way, I'm fine with either of the two names, as long as we don't add the redundant `_config` in the name :)

Reviewer: I associate backoff with backing off in case of errors. I like the promtails naming more. And we can drop the _config suffix of course, I was talking about a high level naming here of concepts.

The `file_watch` block configures how often log files are polled from disk for changes.
The following arguments are supported:

 Name                 | Type       | Description                               | Default | Required 
----------------------|------------|-------------------------------------------|---------|----------
 `min_poll_frequency` | `duration` | Minimum frequency to poll for files.      |  250ms  | no      
 `max_poll_frequency` | `duration` | Maximum frequency to poll for files.      |  250ms  | no       

> Any time file changes are detected, the poll frequency gets reset to `min_poll_frequency`.
> 
> Any time no files changes are detected, the poll frequency doubles in value to the maximum duration specified by `max_poll_frequency`.

Reviewer 2: ```suggestion
If no file changes are detected, the poll frequency doubles until a file change is detected or the poll frequency reaches the `max_poll_frequency`.

If file changes are detected, the poll frequency is reset to `min_poll_frequency`.
```
